channeld/channeld.c

```diff
@@ -669,6 +671,29 @@ static void handle_peer_announcement_signatures(struct peer *peer, const u8 *msg
 				type_to_string(tmpctx, struct channel_id, &chanid));
 	}
 
+	/* BOLT 7:
+	 * - if the node_signature OR the bitcoin_signature is NOT correct:
+	 *  - MAY send a warning and close the connection, or send an error and fail the channel.
+	 *
+	 * In our case, we send an error and stop the open channel procedure. This approach is
+	 * considered overly strict since the peer can recover from it. However, this step is
+	 * optional. If the peer sends it, we assume that the signature must be correct.*/
+	assert(pubkey_from_node_id(&peer_pubkey, &peer->node_ids[REMOTE]));
+	sha256(&peer_announcement_signature, msg, tal_count(msg));
```

Comment:
This is wrong: the message it signs is not the announcement_signatures message, but the announcement message.

This requires a lot more logic!  You need to create the channel_announcement (with towire_channel_announcement, with 4 dummy signatures) and then replicate the logic in heck_channel_announcement() in gossipd/routing.c.

Reply:
Mh my bad I just assumed as a kid. I will check it now thanks!